Runtime/Behaviours/CinemachineConfiner2D.cs

```diff
@@ -0,0 +1,442 @@
+#if !UNITY_2019_3_OR_NEWER
+#define CINEMACHINE_PHYSICS_2D
+#endif
+
+using System;
+using System.Collections.Generic;
+using Cinemachine.Utility;
+using UnityEngine;
+
+namespace Cinemachine
+{
+#if CINEMACHINE_PHYSICS_2D
+    /// <summary>
+    /// An add-on module for Cinemachine Virtual Camera that post-processes the final position of the virtual camera.
+    /// It will confine the virtual camera view window to the area specified in the Bounding Shape 2D field based on
+    /// the camera's window size and ratio. The confining area is baked and cached at start.
+    ///
+    /// 
+    /// CinemachineConfiner2D uses a cache to avoid recalculating the confiner unnecessarily.
+    /// If the cache is invalid, it will be automatically recomputed at the next usage (lazy evaluation).
+    /// The cache is automatically invalidated in some well-defined circumstances:
+    /// - Aspect ratio of the parent vcam changes.
+    /// - MaxOrthoSize parameter in the Confiner changes.
+    ///
+    /// The user can invalidate the cache manually by calling InvalidatePathCache() function or clicking the
+    /// InvalidatePathCache button in the editor on the component.
+    ///
+    /// Collider's Transform changes are supported, but after changing the Rotation component the cache is going to be
+    /// invalid, and the user needs to invalidate it if they want to have a correct cache. If the collider is rotated,
+    /// non-uniform scale will distort the confiner.
+    /// 
+    /// The cache is NOT automatically invalidated (due to high computation cost every frame) if the contents
+    /// of the confining shape change (e.g. points get moved dynamically). In that case, we expose an API to
```

Comment:
Move this paragraph up, and change relevant section to: "in that case, the client must call the InvalidatePathCache() function the user must click the Invalidate Cache button in the component's inspector"